examples/cloudknot_hcp_example.py

```diff
@@ -0,0 +1,183 @@
+    else:
+        seed_mask_obj = afm.FullMask()
+
+    # Determined if n_seeds is per voxel or not
+    if n_seeds > 3:
+        random_seeds = True
+    else:
+        random_seeds = False
+
+    # set the tracking_params based off our inputs
+    tracking_params = {
+        "seed_mask": seed_mask_obj,
+        "n_seeds": n_seeds,
+        "random_seeds": random_seeds}
+
+    # use segmentation file from HCP to get a brain mask,
+    # where everything not labelled 0 is considered a part of the brain
+    brain_mask = afm.LabelledMaskFile(
+        'seg', {'scope': 'dmriprep'}, exclusive_labels=[0])
+
+    # define the api AFQ object
+    myafq = api.AFQ(
+        hcp_bids,
+        brain_mask=brain_mask,
+        tracking_params=tracking_params)
+
+    # export_all runs the entire pipeline and creates many useful derivates
+    myafq.export_all()
+
+    # upload the results to some location on s3
+    myafq.upload_to_s3(
+        s3fs.S3FileSystem(),
+        (f"my_study_bucket/my_study_prefix/derivatives_afq_"
+        f"{seed_mask}_{n_seeds}"))
+
+
+# here we provide a list of subjects that we have selected to process
+subjects = [103818, 105923, 111312]
+
+# here we construct lists of everything we want to test:
+subjects = [str(i) for i in subjects]
+seed_mask = ["fa", "roi"]
+n_seeds = [1, 2, 1000000, 2000000]
+
+# and here we mix the above lists, such that every subject is tried with
+# every mask and every number of seeds
+args = list(itertools.product(subjects, seed_mask, n_seeds))
+
+# Use configparser to get the relevant hcp keys
+# Requires an hcp entry in your ~/.aws/credentials file
+CP = configparser.ConfigParser()
+CP.read_file(open(op.join(op.expanduser('~'), '.aws', 'credentials')))
+CP.sections()
+aws_access_key = CP.get('hcp', 'AWS_ACCESS_KEY_ID')
+aws_secret_key = CP.get('hcp', 'AWS_SECRET_ACCESS_KEY')
+
+# This function will attach your keys to each list in a list of lists
+# We use this with each list being a list of arguments,
+# and we append the aws keys to each list of arguments.
+
+
+def attach_keys(list_of_arg_lists):
+    new_list_of_arg_lists = []
+    for args in list_of_arg_lists:
+        arg_ls = list(args)
+        arg_ls.extend([aws_access_key, aws_secret_key])
+        new_list_of_arg_lists.append(arg_ls)
+    return new_list_of_arg_lists
+
+
+# here we attach the access keys to the argument list
+args = attach_keys(args)
+
+# define the knot to run your jobs on
+# this not bids for access to ec2 resources,
+# so its jobs are cheaper to run but may be evicted
+# installs pyAFQ from github
+knot = ck.Knot(
+    name='afq_hcp_tractography-201110-0',
+    func=afq_process_subject,
+    base_image='python:3.8',
+    image_github_installs="https://github.com/yeatmanlab/pyAFQ.git",
+    pars_policies=('AmazonS3FullAccess',),
+    bid_percentage=100)
+
+# launch a process for each combination
+# Because starmap is True, each list in args will be unfolded
+# and passed into afq_process_subject as arguments
```

Comment:
```suggestion
##########################################################################
# This launches a process for each combination.
# Because `starmap` is `True`, each list in `args` will be unfolded
# and passed into `afq_process_subject` as arguments.
```